File: backend/app/schemas/tiempo_atencion.py
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class TiempoAtencionCreate(BaseModel):
    cliente_id: int | None = None
    comentario_id: int | None = None

    tiempo_minutos: Decimal = Field(
        gt=0
    )

    operador: str | None = None


class TiempoAtencionResponse(BaseModel):
    id: int
    cliente_id: int | None = None
    comentario_id: int | None = None

    tiempo_minutos: Decimal

    fecha: date | None = None
    operador: str | None = None
    created_at: datetime | None = None

    model_config = ConfigDict(
        from_attributes=True
    )

File: frontend/src/services/tiempoAtencion.ts
import { apiFetch } from "./api";


export interface TiempoAtencion {
  id: number;
  cliente_id: number | null;
  comentario_id: number | null;
  tiempo_minutos: number;
  fecha: string | null;
  operador: string | null;
  created_at: string | null;
}


export interface TiempoAtencionCreate {
  cliente_id?: number | null;
  comentario_id?: number | null;
  tiempo_minutos: number;
  operador?: string | null;
}


export function obtenerTiemposAtencion() {
  return apiFetch<TiempoAtencion[]>(
    "/api/tiempos-atencion"
  );
}


export function crearTiempoAtencion(
  datos: TiempoAtencionCreate
) {
  return apiFetch<TiempoAtencion>(
    "/api/tiempos-atencion",
    {
      method: "POST",
      body: JSON.stringify(datos),
    }
  );
}


export function eliminarTiempoAtencion(
  id: number
) {
  return apiFetch<void>(
    `/api/tiempos-atencion/${id}`,
    {
      method: "DELETE",
    }
  );
}